File: src/test/java/mathia/UtilTest.java
package mathia;

import org.apache.commons.math3.geometry.euclidean.twod.Vector2D;
import org.junit.Assert;
import org.junit.Test;

public class UtilTest {
    @Test
    public void linearizedCordToMiddle() throws Exception {
        Vector2D subject;
        subject = Util.linearizedCordToMiddle(4, 3, 1);
        Assert.assertEquals(new Vector2D(1.5f, 1.5f), subject);
        subject = Util.linearizedCordToMiddle(0, 3, 1);
        Assert.assertEquals(new Vector2D(0.5f, 0.5f), subject);
        subject = Util.linearizedCordToMiddle(8, 3, 1);
        Assert.assertEquals(new Vector2D(2.5f, 2.5f), subject);
        subject = Util.linearizedCordToMiddle(3, 3, 1);
        Assert.assertEquals(new Vector2D(0.5f, 1.5f), subject);
    }

    @Test
    public void linearizedCord() throws Exception {
        Assert.assertEquals(4, Util.linearizedCord(1, 1, 3));
        Assert.assertEquals(8, Util.linearizedCord(2, 2, 3));
        Assert.assertEquals(0, Util.linearizedCord(0, 0, 3));
        Assert.assertEquals(1, Util.linearizedCord(1, 0, 3));
        Assert.assertEquals(3, Util.linearizedCord(0, 1, 3));
    }

}

File: src/test/scala/mathia/Util_Test.scala
package mathia

import org.scalatest.FunSuite

class Util_Test extends FunSuite {

  test("testLinearizedNeighbours") {
    val expected: Vector[(Any, Int)] = Vector(
      ((-1, -1), 0), ((0, -1), 1), ((1, -1), 2),
      ((-1, 0), 3), ((1, 0), 5),
      ((-1, 1), 6), ((0, 1), 7), ((1, 1), 8))
    assert(Util.linearizedNeighbours(4, 3, 9) == expected)
  }

}
